### ISSUE TITLE
[Popover]: Improve Accessibility of Popover Component for Screen Readers

### ISSUE THREAD

**Issue by Dannysht** [SAP/ui5-webcomponents-react#7153]
### Describe the bug

The current implementation of the Popover component is not accessible for screen reader users. Screen readers are unable to detect or read the content displayed inside the Popover, which results in a poor user experience for users relying on assistive technologies.

### Isolated Example

https://stackblitz.com/edit/github-o8ow4gct?file=src%2FApp.tsx

### Reproduction steps

1.
2.
3.
...

### Expected Behaviour

When the Popover is opened, screen readers should be notified and able to read its content in full.

The Popover content should be semantically structured and navigable (e.g., sections, headings, forms, lists).

Focus should be properly managed when the Popover opens and closes (e.g., focus is trapped inside while open and returns to the triggering element on close).

Appropriate ARIA roles and attributes should be used to convey the role and structure of the Popover content.

### UI5 Web Components for React Version

2.8

### UI5 Web Components Version

2.8

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.
